vortex-serde/src/file/reader/batch.rs

```diff
@@ -0,0 +1,69 @@
+use std::collections::{HashMap, VecDeque};
+use std::sync::Arc;
+
+use vortex::array::struct_::StructArray;
+use vortex::{Array, IntoArray};
+use vortex_dtype::DType;
+use vortex_error::VortexResult;
+
+use super::schema::Schema;
+use crate::file::layouts::Layout;
+use crate::file::reader::column::ColumnReader;
+use crate::io::VortexReadAt;
+
+pub(super) struct BatchReader<R> {
+    readers: HashMap<Arc<str>, ColumnReader>,
```

Comment:
I think this could just be a `Vec<ColumnReader>` since we always construct it by iterating over schema. You actually don't even need the schema here but just `Vec<(Arc<str>, ColumnReader)>`

Reply:
Done